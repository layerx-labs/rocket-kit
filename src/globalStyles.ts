import { createGlobalStyle } from 'styled-components/macro';
import { rem} from "polished";
import { typography, colors } from './ions/variables';
import { device } from './ions/breakpoints';

const GlobalStyle = createGlobalStyle`
  * {
    box-sizing: border-box;
  }

  html {
    font-size: 16px;
    scroll-behavior: smooth;
  }

  body {
    margin: 0;
    font-family: 'Space Grotesk', Verdana, Arial, Helvetica, sans-serif;
    font-size: 1rem;
    font-weight: ${typography.regular};
    color: ${colors.black};
    font-feature-settings: 'ss04' on;
    -webkit-font-smoothing: antialiased;
  }

  h1,
  h1 span,
  h2,
  h2 span {
    margin: 0;
    font-weight: ${typography.regular};
  }

  h1,
  h1 span {
    font-family: 'Laforgestencil', Verdana, Arial, Helvetica, sans-serif;
    font-size: 3.5rem;
    line-height: 1.15;
  }

  h2,
  h2 span {
    font-size: 2rem;
    line-height: 1.2;
  }

  h3,
  h4 {
    margin: 0;
    font-size: 1rem;
    font-weight: 400;
  }

  span {
    font-family: ${typography.defaultFont};
    font-size: ${rem(typography.defaultSize)};
    font-feature-settings: 'ss04' on;
  }

  p {
    margin: 0;
    font-size: 1rem;
    line-height: 1.3;

    &:not(:last-child) {
      margin-bottom: ${rem('10px')};
    }
  }

  pre {
    border: 1px solid ${colors.grey500};
    border-radius: 4px;
    background-color: ${colors.grey300};
    padding: ${rem('15px')};
  }

  @media ${device.l} {
    h1,
    h1 span {
      font-size: 6rem;
      line-height: 1.1;
    }

    h2,
    h2 span {
      font-size: 2.5rem;
    }
  }
`;

export default GlobalStyle;
