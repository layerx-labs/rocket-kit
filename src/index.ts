// Atoms
export { default as Avatar } from './atoms/avatar';
export { default as Button } from './atoms/button';
export { default as ButtonLink } from './atoms/button-link';
export { default as Checkbox } from './atoms/checkbox';
export { default as Icon } from './atoms/icon';
export { default as Spinner } from './atoms/spinner';
export { default as Tag } from './atoms/tag';

// Molecules
export { default as NoteCard } from './molecules/note-card';
