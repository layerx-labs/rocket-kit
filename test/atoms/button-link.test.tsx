import React from 'react';
import { render, screen } from '@testing-library/react';
import { ButtonLink } from '../../src';

describe('Button Link', () => {
  it('renders', () => {
    const { asFragment } = render(
      <ButtonLink url="https://github.com/taikai/taikai-design-system" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Button has correct value', async () => {
    const buttonText = 'Check the source code';
    render(
      <ButtonLink
        url="https://github.com/taikai/taikai-design-system"
        value={buttonText}
      />
    );
    const foundText = await screen.getByText(buttonText);
    expect(foundText).toBeTruthy();
  });

  it('Button has correct href', async () => {
    const buttonText = 'Check the source code';
    const buttonUrl = 'https://github.com/taikai/taikai-design-system';
    render(<ButtonLink url={buttonUrl} value={buttonText} />);
    const anchorElem = await screen.getByText(buttonText).closest('a');
    expect(anchorElem).toHaveProperty('href', buttonUrl);
  });

  it('Button has target _blank and noreferrer', async () => {
    const buttonText = 'Check the source code';
    const buttonUrl = 'https://github.com/taikai/taikai-design-system';
    render(<ButtonLink url={buttonUrl} value={buttonText} blank />);
    const anchorElem = await screen.getByText(buttonText).closest('a');
    expect(anchorElem).toHaveProperty('target', '_blank');
    expect(anchorElem).toHaveProperty('rel', 'noopener noreferrer');
  });

  it('Button has empty targe if not defined', async () => {
    const buttonText = 'Check the source code';
    const buttonUrl = 'https://github.com/taikai/taikai-design-system';
    render(<ButtonLink url={buttonUrl} value={buttonText} />);
    const anchorElem = await screen.getByText(buttonText).closest('a');
    expect(anchorElem).toHaveProperty('target', '');
  });
});
