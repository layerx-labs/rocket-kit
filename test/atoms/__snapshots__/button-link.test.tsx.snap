// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Link renders 1`] = `
<DocumentFragment>
  <a
    class="sc-dlfnbm gWdZLC button button-link"
    color="primary"
    data-event=""
    data-testid=""
    href="https://github.com/taikai/taikai-design-system"
    rel=""
    target=""
    value=""
  />
</DocumentFragment>
`;
